plugins/woocommerce-admin/client/products/sections/images-section.tsx

```diff
@@ -0,0 +1,191 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Link,
+	useFormContext,
+	MediaUploader,
+	ImageGallery,
+	ImageGalleryItem,
+} from '@woocommerce/components';
+import { Card, CardBody, DropZone } from '@wordpress/components';
+import { recordEvent } from '@woocommerce/tracks';
+import { useState } from '@wordpress/element';
+import { Product } from '@woocommerce/data';
+import classnames from 'classnames';
+import { Icon, trash } from '@wordpress/icons';
+import { MediaItem } from '@wordpress/media-utils';
+
+/**
+ * Internal dependencies
+ */
+import { ProductSectionLayout } from '../layout/product-section-layout';
+import DragAndDrop from '../images/drag-and-drop.svg';
+import './images-section.scss';
+
+type Image = MediaItem & {
+	src: string;
+};
+
+export const ImagesSection: React.FC = () => {
+	const { getInputProps, setValue } = useFormContext< Product >();
+	const images = ( getInputProps( 'images' ).value as Image[] ) || [];
+	const [ showRemoveZone, setShowRemoveZone ] = useState( false );
+	const [ imageIdToRemove, setImageIdToRemove ] = useState< number >( 0 );
+
+	const getUniqueImages = ( files: Image[] ) => {
+		if ( ! files ) {
+			return [];
+		}
+		files.forEach( ( image: Image ) => {
+			if (
+				image.id &&
+				images.find( ( file ) => file.id === image.id ) === undefined
+			) {
+				images.push( image );
+			}
+		} );
+		setValue( 'images', images );
+		return images;
+	};
+
+	const toggleRemoveZone = () => {
+		setShowRemoveZone( ! showRemoveZone );
+	};
+
+	const setImageToRemove = ( image: string ) => {
+		const tempNode = document.createElement( 'div' );
+		tempNode.innerHTML = image;
+		const imageId = tempNode.getElementsByTagName( 'img' )[ 0 ].id;
```

Comment:
I think instead of retrieving this ID via selectors, we could set the ID of the dragged element on start.  For example:

```
onDragStart={ ( image ) => setDraggedImageId( image.key ) }
```

Reply:
I'm not sure if I understood this suggestion correctly. When a user grabs an image, we don't know if the user is wanting to modify the order of the images or remove that image. If the user wants only to change the order of the images, we wouldn't need to save that value.

Reply:
Got it! I like the idea! I changed it in commit [ddccb5b](https://github.com/woocommerce/woocommerce/pull/34769/commits/ddccb5b3983c05b44da6b6335ad8f888d8511fdc)